### DISCUSSION TITLE
How to stream headers with undici.stream?

### DISCUSSION THREAD

**Discussion by wlinna** [nodejs/undici#2919] [Q&A] [answered]
Hello, I am trying to follow this example. My endpoint is practically a proxy, and I need it to be as efficient as possible.

https://github.com/nodejs/undici/blob/main/docs/docs/api/Dispatcher.md#example-2---stream-to-fastify-response

The relevant code in my fastify handler is here
```
            undici.stream(addr, {
                method: 'GET',                
                opaque: res
            }, factoryData => (<any>factoryData.opaque).raw);
            return res;
```

However, it seems that the headers are not streamed to the response, and I don't see a way to fix it. What can I do?

**Comment by metcoder95** [comment]
Can you provide an 
[Minimum Reproducible Example](https://stackoverflow.com/help/minimal-reproducible-example)?

From here it seems you are not attaching the headers to your `undici.stream` call.

**Reply by wlinna** [reply, marked as answer]
> From here it seems you are not attaching the headers to your undici.stream call.

Now I see how blind I've been. [Example 1](https://github.com/nodejs/undici/blob/main/docs/docs/api/Dispatcher.md#example-1---basic-get-stream-request) shows that `factory` receives `headers` parameter that I can use to add the headers to the stream.

This is what my relevant code (TypeScript) looks like now, and it seems to work

```
            undici.stream(addr, {
                method: 'GET',
                opaque: res
            }, ({ headers, opaque }) => {
                const raw = (<any>opaque).raw;
                if (headers) {
                    for (const key in headers) {
                        const val = headers[key];
                        if (val) {
                            raw.appendHeader(key, val);
                        }
                    }
                }
                return raw;
            });
```
